### ISSUE TITLE
Broken assertSame/assertNotSame after #2460

### ISSUE THREAD

**Issue by Axinet** [cbeust/testng#2616]
### Expected behavior
assertNotSame(object1, object2) should pass, when `object1 != object2 && object1.equals(object2)`

### Actual behavior
After merging https://github.com/cbeust/testng/commit/841ae94bd07f68695878f679e03f6d6be6a1884d (issue #2460) it looks like replacing `actual==expected` / `actual!=expected` to `equals`/ `!equals` went too far and broken one of common Assetions.  Its a strong **backward incompatible change in behaviour between 7.3 and 7.4**

### Is the issue reproductible on runner?

(Probably everywhere, tested on two)
- [ ] Shell
- [X] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [X] IntelliJ
- [ ] NetBeans

### Test case sample
> Please, share the test case (as small as possible) which shows the issue

Problem is present for any custom object which has implemented its own equals method.

```
import java.util.Objects;

import org.testng.Assert;
import org.testng.annotations.Test;

public class BrokenNotSameTest {
	static class MyObject {
		private final String myName;

		MyObject(String myName) {
			this.myName = myName;
		}

		@Override
		public boolean equals(Object o) {
			if (this == o) {
				return true;
			}
			if (o == null || getClass() != o.getClass()) {
				return false;
			}
			MyObject myObject = (MyObject) o;
			return Objects.equals(myName, myObject.myName);
		}

		@Override
		public int hashCode() {
			return Objects.hash(myName);
		}
	}

	@Test
	public static void assertNotSamePass73Fails74() {
		MyObject instance1 = new MyObject("someName");
		MyObject instance2 = new MyObject("someName");
		Assert.assertNotSame(instance1, instance2);
	}

	@Test
	public static void assertSameInstancesFails73Pass74() {
		MyObject instance1 = new MyObject("someName");
		MyObject instance2 = new MyObject("someName");
		Assert.assertSame(instance1, instance2);
	}
}
```

**Comment by juherr** [comment]
Duplicate of #2486
